substrate/frame/support/test/tests/runtime.rs

```diff
@@ -256,24 +281,64 @@ pub type Header = generic::Header<BlockNumber, BlakeTwo256>;
 pub type UncheckedExtrinsic = generic::UncheckedExtrinsic<u32, RuntimeCall, Signature, ()>;
 pub type Block = generic::Block<Header, UncheckedExtrinsic>;
 
-frame_support::construct_runtime!(
-	pub struct Runtime
-	{
-		System: frame_system::{Pallet, Call, Event<T>, Origin<T>} = 30,
-		Module1_1: module1::<Instance1>::{Pallet, Call, Storage, Event<T>, Origin<T>},
-		Module2: module2::{Pallet, Call, Storage, Event<T>, Origin},
-		Module1_2: module1::<Instance2>::{Pallet, Call, Storage, Event<T>, Origin<T>},
-		NestedModule3: nested::module3::{Pallet, Call, Config<T>, Storage, Event<T>, Origin},
-		Module3: self::module3::{Pallet, Call, Config<T>, Storage, Event<T>, Origin<T>},
-		Module1_3: module1::<Instance3>::{Pallet, Storage, Event<T> } = 6,
-		Module1_4: module1::<Instance4>::{Pallet, Call, Event<T> } = 3,
-		Module1_5: module1::<Instance5>::{Pallet, Event<T>},
-		Module1_6: module1::<Instance6>::{Pallet, Call, Storage, Event<T>, Origin<T>} = 1,
-		Module1_7: module1::<Instance7>::{Pallet, Call, Storage, Event<T>, Origin<T>},
-		Module1_8: module1::<Instance8>::{Pallet, Call, Storage, Event<T>, Origin<T>} = 12,
-		Module1_9: module1::<Instance9>::{Pallet, Call, Storage, Event<T>, Origin<T>},
-	}
-);
+#[frame_support::runtime]
+mod runtime {
+	#[runtime::runtime]
+	#[runtime::derive(
+		RuntimeCall,
+		RuntimeEvent,
+		RuntimeError,
+		RuntimeOrigin,
+		RuntimeFreezeReason,
+		RuntimeHoldReason,
+		RuntimeSlashReason,
+		RuntimeLockId,
+		RuntimeTask
+	)]
+	pub struct Runtime;
+
+	#[runtime::pallet_index(30)]
+	pub type System = frame_system + Pallet + Call + Event<T> + Origin<T>;
+
+	#[runtime::pallet_index(31)]
+	pub type Module1_1 = module1<Instance1>;
+
+	#[runtime::pallet_index(32)]
+	pub type Module2 = module2;
+
+	#[runtime::pallet_index(33)]
+	pub type Module1_2 = module1<Instance2>;
+
+	#[runtime::pallet_index(34)]
+	pub type NestedModule3 = nested::module3;
+
+	#[runtime::pallet_index(35)]
+	#[runtime::disable_unsigned]
+	pub type Module3 = self::module3;
+
+	#[runtime::pallet_index(6)]
+	#[runtime::disable_call]
+	pub type Module1_3 = module1<Instance3>;
+
+	#[runtime::pallet_index(3)]
+	pub type Module1_4 = module1<Instance4>;
+
+	#[runtime::pallet_index(4)]
+	#[runtime::disable_call]
+	pub type Module1_5 = module1<Instance5>;
+
+	#[runtime::pallet_index(1)]
+	pub type Module1_6 = module1<Instance6>;
+
+	#[runtime::pallet_index(2)]
+	pub type Module1_7 = module1<Instance7>;
+
+	#[runtime::pallet_index(12)]
+	pub type Module1_8 = module1<Instance8>;
```

Comment:
Are `cfg` attributes supported here?  
IMHO i think its not a good idea, but *someone* will most likely try it 😆

Reply:
Not at this point, but I can add as a follow-up?

Reply:
Yea should probably be recorded as follow up issue.

Reply:
Created this tracking issue: https://github.com/paritytech/polkadot-sdk/issues/3688